examples/fluids/src/petsc_ops.c

```diff
@@ -35,49 +50,57 @@ PetscErrorCode OperatorApplyContextCreate(DM dm_x, DM dm_y, Ceed ceed, CeedOpera
     }
   }
 
-  PetscCall(PetscNew(op_apply_ctx));
+  PetscCall(PetscNew(ctx));
 
   // Copy PETSc Objects
   if (dm_x) PetscCall(PetscObjectReference((PetscObject)dm_x));
-  (*op_apply_ctx)->dm_x = dm_x;
+  (*ctx)->dm_x = dm_x;
   if (dm_y) PetscCall(PetscObjectReference((PetscObject)dm_y));
-  (*op_apply_ctx)->dm_y = dm_y;
+  (*ctx)->dm_y = dm_y;
 
   if (X_loc) PetscCall(PetscObjectReference((PetscObject)X_loc));
-  (*op_apply_ctx)->X_loc = X_loc;
+  (*ctx)->X_loc = X_loc;
   if (Y_loc) PetscCall(PetscObjectReference((PetscObject)Y_loc));
-  (*op_apply_ctx)->Y_loc = Y_loc;
+  (*ctx)->Y_loc = Y_loc;
 
   // Copy libCEED objects
-  if (x_ceed) CeedVectorReferenceCopy(x_ceed, &(*op_apply_ctx)->x_ceed);
-  if (y_ceed) CeedVectorReferenceCopy(y_ceed, &(*op_apply_ctx)->y_ceed);
-  CeedOperatorReferenceCopy(op_apply, &(*op_apply_ctx)->op);
-  CeedReferenceCopy(ceed, &(*op_apply_ctx)->ceed);
+  if (x_ceed == CEED_VECTOR_NONE) (*ctx)->x_ceed = x_ceed;
```

Comment:
I figured `CEED_VECTOR_NONE` is a bit more explicit about intent. Note though that `NULL` actually works just fine. `CeedOperator{Add}Apply` doesn't seem to care between `CEED_VECTOR_NONE` and `NULL`. Documentation-wise, it's supposed to take `CEED_VECTOR_NONE` if there are no active inputs. I think maybe we should have it be exclusively `CEED_VECTOR_NONE` and error if it's passed a `NULL` active vector.

Reply:
> I figured `CEED_VECTOR_NONE` is a bit more explicit about intent.

ie. If I pass `CEED_VECTOR_NONE`, then I'm going to using the `Local` apply function call and pass it a `NULL` local `Vec`:
```c
  PetscCall(OperatorApplyContextCreate(NULL, NULL, ceed, op_quad_coords, CEED_VECTOR_NONE, NULL, NULL, NULL, &op_quad_coords_ctx));
  PetscCall(ApplyCeedOperatorLocalToLocal(NULL, Output_loc, op_quad_coords_ctx));
```

If I pass `NULL`, then I expect that the CeedVector be created on context creation to then be as the container for the local `Vec` I pass to the apply function:
```c
  PetscCall(OperatorApplyContextCreate(NULL, NULL, ceed, op_quad_coords, NULL, NULL, NULL, NULL, &op_quad_coords_ctx));
  PetscCall(ApplyCeedOperatorLocalToLocal(Input_loc, Output_loc, op_quad_coords_ctx));
```

Reply:
Interesting. We don't have a scenario in Ratel in which there is no input Lvec. It might make sense for clarity to have `ApplyCeedOperatorPassiveToLocal`? 

Could 
```
 if (y_ceed) CeedVectorReferenceCopy(y_ceed, &(*ctx)->y_ceed);
 else if (y_size > 0) CeedVectorCreate(ceed, y_size, &(*ctx)->y_ceed);
```

Its probably a good idea to make `CeedVectorReferenceCopy` work for `CEED_VECTOR_ACTIVE` and `CEED_VECTOR_NONE` (and `CeedVectorDestroy)`.

> I think maybe we should have it be exclusively `CEED_VECTOR_NONE` and error if it's passed a NULL active vector.

Currently we allow `NULL` as an input as a holdover from before `CEED_VECTOR_NONE` existed.

Reply:
See https://github.com/CEED/libCEED/pull/1186

Reply:
> > I think maybe we should have it be exclusively `CEED_VECTOR_NONE` and error if it's passed a NULL active vector.
> 
> Currently we allow `NULL` as an input as a holdover from before `CEED_VECTOR_NONE` existed.

Gotcha. I'm assuming by holdover, you mean keeping it for backwards compatibility reasons? Otherwise, I think it makes sense to make the user deliberately choose `CEED_VECTOR_NONE` rather than it take `NULL`. 

> It might make sense for clarity to have `ApplyCeedOperatorPassiveToLocal`?

Maybe. I'll have a think on it, as I think `ApplyCeedOperatorLocalToLocal(NULL, Y_loc, ctx);` could work just as well. Currently there's a "backdoor" way to have a `CeedVector` active field; if `NULL` is passed for `X/Y_loc`, it'll just use whatever `ctx->x_ceed` has, which could be an actual `CeedVector` if desired. That backdoor will need to be cutoff for `ApplyCeedOperatorLocalToLocal(NULL,...)` to have the same behavior as a `ApplyCeedOperatorPassiveToLocal`.

Reply:
> Gotcha. I'm assuming by holdover, you mean keeping it for backwards compatibility reasons?

Yeah, that's why it stuck around for so long. Might be interesting to see in a separate PR what breaks if we remove this.